packages/flutter_tools/lib/src/android/gradle_errors.dart

```diff
@@ -669,3 +670,47 @@ final GradleHandledError sslExceptionHandler = GradleHandledError(
   },
   eventLabel: 'ssl-exception-tag-mismatch',
 );
+
+/// If an incompatible Java and Gradle versions error is caught, we expect an
+/// error specifying that the Java major class file version, one of
+/// https://javaalmanac.io/bytecode/versions/, is unsupported by Gradle.
+final RegExp _unsupportedClassFileMajorVersionPattern = RegExp(r'Unsupported class file major version\s+([0-9]{2})');
```

Comment:
Is this always 2 digits?

Reply:
Maybe the following would be more robust:
```suggestion
final RegExp _unsupportedClassFileMajorVersionPattern = RegExp(r'Unsupported class file major version\s+([0-9]+)');
```

Reply:
Won't it eventually overflow when we reach version 100?

Reply:
Also, what happened to the first 44 bytecode versions lol

Reply:
Lol unclear, but yeah I think it's worth the change